### DISCUSSION TITLE
Battery switches/remotes with zigbee binding, lights adapt some seconds after turn on

### DISCUSSION THREAD

**Discussion by sindreruud** [basnijholt/adaptive-lighting#1065] [Q&A]
Hi!

I've seen this question popped up in several issues and discussions. But I can't seem to quite wrap my head around it. Some people say it is not possible, while other come with solutions. But the solutions doesn't seem to quite fit my usecase. So hopefully someone here can tell me if what I want can even be achieved. 

I have Home Assistant, Zigbee2MQTT addon and the MQTT broker. 
Currently I am just getting started, so my house is not complicated at all. It consists of two hue lights (a bulb and a led strip) which is always powered, and I use the Hue Remote to turn them off/on. They are all in z2m, I have removed the Hue hub.

The two lights are in a zigbee group that I made in z2m. The Hue Remote has a zigbee binding to this group. My understanding is that this allows the Hue Remote to turn on/off/dim the lights by talking directly to them, without the need to go via HA/z2m. This I find rather important, as it makes the experience of using the remote more snappy, and it will also continue working even if HA/z2m is down for some reason.

Now over to adaptive lighting. I have this configured to adapt brightness and color temperature, and it is working really well if I only use the Home Assistant app. Turning on lights in the app, they turn on to the correct brightness and color temperature. I have not had problems with lights randomly turning off or on, or not adapting. It's just working.

So my only issue is when turning on lights with the Hue Remote. The lights turn on to the last state they were in, both brightness and color temperature. And then within some seconds, it seems like adaptive lighting kicks in and adjusts them to the correct brightness and color temperature. This is very noticable if I turn of the lights at 13:00 when they are peak brightness and color temp, and then I come back home at 22:00 and turn them on. You get blinded for a couple of seconds untill the adaption corrects the brightness/temp

Is this a limition with the Hue remote, or zigbee bindings or is it possible to solve this? I am willing to buy a new switch/remote if that would solve it, or use two switches where one is bound to the controller and the other to the lights.

**Comment by moemeli** [comment]
Hello! 
I think this is because your lights send their state to the controller when they turn on after button press on the remote and adaptive lighting does not know they are on before that happens and this is causing the delay. I am using some lights just like that, but with the exception that I am using ZHA. I still think Z2M will act the same and you should be able to do it in the same way.

I have binded the remote like you to the zigbee lightgroup, but it is also connected to the controller. So if the controller is down, it will still use direct binding to turn on the lights (I have tested it works). Then from HA side I have made an automation that will trigger adaptive lighting to adapt the lights when the corresponding button is pressed on the remote.

Now there is a down side with the newer hue remotes: the on/off toggle button on the remote does not know if the lights are on or not, it only knows the last command it sent to the lights. So I am using long press on the on/off toggle to trigger this automation. (so that it does not trigger when I want the lights to turn off)

Hope this helps a bit!
